tests/system/python/packages/test_north_pi_webapi_nw_throttle.py

```diff
@@ -0,0 +1,272 @@
+    tracked_item = tracking_details["track"][0]
+    assert ASSET == tracked_item["asset"]
+    assert "coap" == tracked_item["plugin"]
+
+    if not skip_verify_north_interface:
+        egress_tracking_details = utils.get_asset_tracking_details(fledge_url, "Egress")
+        assert len(egress_tracking_details["track"]), "Failed to track Egress event"
+        tracked_item = egress_tracking_details["track"][0]
+        assert ASSET == tracked_item["asset"]
+        assert "OMF" == tracked_item["plugin"]
+
+
+def _verify_egress(read_data_from_pi_web_api, pi_host, pi_admin, pi_passwd, pi_db, wait_time, retries, asset_name):
+    retry_count = 0
+    data_from_pi = None
+
+    af_hierarchy_level = "fledge/room1/machine1"
+    af_hierarchy_level_list = af_hierarchy_level.split("/")
+    type_id = 1
+    dp_name = 'id_datapoint'
+    recorded_datapoint = "{}measurement_{}.{}".format(type_id, asset_name, dp_name)
+
+    # Name of asset in the PI server
+    PI_ASSET_NAME = "{}-type{}".format(asset_name, type_id)
+
+    while (data_from_pi is None or data_from_pi == []) and retry_count < retries:
+        data_from_pi = read_data_from_pi_web_api(pi_host, pi_admin, pi_passwd, pi_db, af_hierarchy_level_list,
+                                                 PI_ASSET_NAME, {recorded_datapoint})
+        retry_count += 1
+        time.sleep(wait_time * 2)
+
+    if data_from_pi is None or retry_count == retries:
+        assert False, "Failed to read data from PI"
+
+    return data_from_pi
+
+
+@pytest.fixture
+def start_south_north(add_south, start_north_task_omf_web_api, add_filter, remove_data_file,
+                      fledge_url, pi_host, pi_port, pi_admin, pi_passwd,
+                      start_north_omf_as_a_service, start_north_as_service, asset_name=ASSET):
+    """ This fixture
+        clean_setup_fledge_packages: purge the fledge* packages and install latest for given repo url
+        add_south: Fixture that adds a south service with given configuration
+        start_north_task_omf_web_api: Fixture that starts PI north task
+        remove_data_file: Fixture that remove data file created during the tests """
+
+    south_plugin = "sinusoid"
+    # south_branch does not matter as these are archives.fledge-iot.org version install
+    _config = {"assetName": {"value": ASSET}}
+    add_south(south_plugin, None, fledge_url, config=_config,
+              service_name=SOUTH_SERVICE_NAME, installation_type='package')
+    if not start_north_as_service:
+        start_north_task_omf_web_api(fledge_url, pi_host, pi_port, pi_user=pi_admin, pi_pwd=pi_passwd,
+                                     start_task=False, taskname=NORTH_TASK_NAME)
+    else:
+        start_north_omf_as_a_service(fledge_url, pi_host, pi_port, pi_user=pi_admin, pi_pwd=pi_passwd,
+                                     start=False, service_name=NORTH_TASK_NAME)
+
+    add_filter("python35", None, "py35", {}, fledge_url, None, installation_type='package',
```

Comment:
Instead of adding flag and then adding filter + pipeline to north instance; can not we use add_filter as is and just upload the script?

Reply:
We should not start the north instance before applying filter. Otherwise we will not be able to insert reading  id from the very first reading itself.  Also I wanted to install filter only. Then we need to upload a script to the filter. This is a very specific thing we have to do. So we cannot utilize the existing fixture add_filter.